### ISSUE TITLE
Rendering of bitmaps

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1694]
**Reported by hansolsson on 10 Apr 2015 14:14 UTC**
In 18.6.5.6 Bitmap
It is stated that:

Given an extent {{x1, y1}, {x2, y2}}, x2<x1 defines horizontal flipping and y2<y1 defines vertical flipping around the point defined by the origin attribute.

One would then assume that if x1<x2 and y1<y2 the bitmap would look the same as if opened in e.g. paint.

However, looking at trunk most bitmaps don't have unflipped coodinates, and many would look odd if rendering with this interpretation of the specification. The following cases were found (I only checked that coordinates are flipped and that it looks ok if unflipped):

Modelica.Mechanics.MultiBody.Examples.Loops.Utilities.EngineV6_analytic.
Modelica.Mechanics.MultiBody.Examples.Systems.RobotR3.Components.MechanicalStructure.
Modelica.Mechanics.MultiBody.Visualizers.Torus.
Modelica.Mechanics.MultiBody.Visualizers.VoluminousWheel.
Modelica.Mechanics.MultiBody.Visualizers.PipeWithScalarField.
Modelica.Mechanics.MultiBody.Visualizers.Advanced.PipeWithScalarField.
Modelica.Mechanics.Rotational.Examples.Utilities.DirectInertia.
Modelica.Mechanics.Rotational.Examples.Utilities.InverseInertia.
Modelica.Mechanics.Rotational.Examples.Utilities.SpringDamper.
Modelica.Mechanics.Rotational.Examples.Utilities.Spring.
Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.Channel_i.
Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.General_i.
Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HeatExchanger_i.
Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HelicalPipe_i.
Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.Plate_i.
Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.StraightPipe_i.
Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Bend_i.
Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Channel_i.
Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.General_i.
Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.HeatExchanger_i.
Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Orifice_i.
Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.StraightPipe_i.
Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Valve_i.

Assuming we don't intend to change the specification I could automatically change these models to ensure that the extent is given in ordered form.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1694

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 22 Jun 2015 11:50 UTC**
Fixed in b920bad9cf3505b60e97872e20ac056ac409b935 for maintenance and in 963130ce902f63c0257910b65baa23ba23a2b09f for trunk.

The provided Dymola script performing these changes gave the following information:

```
// Only for maintenance:
Converting name LinePattern.None to LinePattern.Solid in Modelica.Electrical.QuasiStationary.MultiPhase.Ideal.IdealCommutingSwitch.
Converting name LinePattern.None to LinePattern.Solid in Modelica.Electrical.QuasiStationary.MultiPhase.Ideal.IdealIntermediateSwitch.

// For maintenance and trunk:
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Examples.Loops.Utilities.EngineV6_analytic.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Examples.Systems.RobotR3.Components.MechanicalStructure.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.Torus.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.VoluminousWheel.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.PipeWithScalarField.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.Advanced.PipeWithScalarField.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.DirectInertia.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.InverseInertia.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.SpringDamper.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.Spring.
Converting name LinePattern.None to LinePattern.Solid in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HelicalPipe1_d.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.Channel_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.General_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HeatExchanger_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HelicalPipe_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.Plate_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.StraightPipe_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Bend_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Channel_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.General_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.HeatExchanger_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Orifice_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.StraightPipe_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Valve_i.
```

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 22 Jun 2015 11:50 UTC**
Bitmap annotations corrected

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 22 Jun 2015 12:05 UTC**
I think there are several issues with the introduced fixes and they should be reverted at once. And then applied cleanly to **only** the classes that really contain fixes. Also it seems they introduced errors:

```diff
- connect(internalSupportR.flange, fixedR.flange) annotation (Line(
+ connect(internalSupportR.flange, fixedR.flange) annotation ={Line(
```

does not look like a legal Modelica change.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 22 Jun 2015 12:09 UTC**
Replying to [comment:2 dietmarw]:
> I think there are several issues with the introduced fixes and they should be reverted at once.

Yes, I also detected this and reverted the change. Obviously, there is an error in the script that I used.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 22 Jun 2015 12:15 UTC**
Replying to [comment:3 otter]:
> Replying to [comment:2 dietmarw]:
> > I think there are several issues with the introduced fixes and they should be reverted at once.
> 
> Yes, I also detected this and reverted the change. Obviously, there is an error in the script that I used.

Yes, I found the error.

It was an error related to something else that caused this problem. 
It has now been corrected; it's just a matter of redoing it.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 22 Jun 2015 12:39 UTC**
And please only commit changes to the packages that have a real change and not all the others where Dymola decided to break a line at yet another place.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 22 Jun 2015 12:56 UTC**
Fixed in f369802d4794f0cf4ce0aa8db64541c36833fb8b for maintenance and in 2fbca7f43cb7ac31921ecaaf20c92ec3f019620f for trunk.

```
// Only for maintenance:
Converting name LinePattern.None to LinePattern.Solid in Modelica.Electrical.QuasiStationary.MultiPhase.Ideal.IdealCommutingSwitch.
Converting name LinePattern.None to LinePattern.Solid in Modelica.Electrical.QuasiStationary.MultiPhase.Ideal.IdealIntermediateSwitch.

// For maintenance and trunk:
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Examples.Loops.Utilities.EngineV6_analytic.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Examples.Systems.RobotR3.Components.MechanicalStructure.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.Torus.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.VoluminousWheel.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.PipeWithScalarField.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.MultiBody.Visualizers.Advanced.PipeWithScalarField.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.DirectInertia.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.InverseInertia.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.SpringDamper.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Mechanics.Rotational.Examples.Utilities.Spring.
Converting name LinePattern.None to LinePattern.Solid in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HelicalPipe1_d.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.Channel_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.General_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HeatExchanger_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.HelicalPipe_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.Plate_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.HeatTransfer.StraightPipe_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Bend_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Channel_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.General_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.HeatExchanger_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Orifice_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.StraightPipe_i.
Converting name Bitmap.extent to Bitmap.extent - ordered in Modelica.Fluid.Dissipation.Utilities.Icons.PressureLoss.Valve_i.
```
